controllers/cluster_controller.go

```diff
@@ -570,15 +571,12 @@ func aggregatedGeneration(config *c.Config) int64 {
 }
 
 func getManagementCluster(ctx context.Context, clus *anywherev1.Cluster, client client.Client) (*anywherev1.Cluster, error) {
-	mgmtCluster := &anywherev1.Cluster{}
-	if err := client.Get(ctx, types.NamespacedName{Name: clus.ManagedBy(), Namespace: clus.Namespace}, mgmtCluster); err != nil {
-		if apierrors.IsNotFound(err) {
-			failureMessage := fmt.Sprintf("Management cluster %s does not exist", clus.Spec.ManagementCluster.Name)
-			clus.SetFailure(anywherev1.ManagementClusterRefInvalidReason, failureMessage)
-		}
+	mgmtCluster, err := clusters.FetchManagementEksaCluster(ctx, client, clus)
+	if err != nil {
+		failureMessage := fmt.Sprintf("Management cluster %s does not exist", clus.Spec.ManagementCluster.Name)
```

Comment:
this is not necessarily true, right? it could a transient error, not always a not found